awswrangler/athena/_statements.py

```diff
@@ -0,0 +1,165 @@
+    workgroup : str, optional
+        The name of the workgroup to which the prepared statement belongs.
+    mode: str
+        Determines the behaviour if the prepared statement already exists:
+
+        - ``update`` - updates statement if already exists
+        - ``error`` - throws an error if table exists
+    boto3_session : boto3.Session(), optional
+        Boto3 Session. The default boto3 session will be used if boto3_session receive None.
+
+    Examples
+    --------
+    >>> import awswrangler as wr
+    >>> res = wr.athena.prepare_statement(
+    ...     sql="SELECT * FROM my_table WHERE name = ?",
+    ...     statement_name="statement",
+    ... )
+    """
+    if mode not in ["update", "error"]:
+        raise exceptions.InvalidArgumentValue("`mode` must be one of 'update' or 'error'.")
+
+    athena_client = _utils.client("athena", session=boto3_session)
+    workgroup = workgroup if workgroup else "primary"
+
+    already_exists = _does_statement_exist(statement_name, workgroup, athena_client)
+    if already_exists and mode == "error":
+        raise exceptions.AlreadyExists(f"Prepared statement {statement_name} already exists.")
+
+    if already_exists:
+        _logger.info(f"Updating prepared statement {statement_name}")
+        athena_client.update_prepared_statement(
+            StatementName=statement_name,
+            WorkGroup=workgroup,
+            QueryStatement=sql,
+        )
+    else:
+        _logger.info(f"Creating prepared statement {statement_name}")
+        athena_client.create_prepared_statement(
+            StatementName=statement_name,
+            WorkGroup=workgroup,
+            QueryStatement=sql,
+        )
+
+
+@apply_configs
+def list_prepared_statements(
+    workgroup: Optional[str] = None, boto3_session: Optional[boto3.Session] = None
+) -> List[str]:
+    """
+    List the prepared statements in the specified workgroup.
+
+    Parameters
+    ----------
+    workgroup: str, optional
+        The name of the workgroup to which the prepared statement belongs.
+    boto3_session : boto3.Session(), optional
+        Boto3 Session. The default boto3 session will be used if boto3_session receive None.
+
+    Returns
+    -------
+    List[Dict[str, Any]]
+        List of prepared statements in the workgroup.
+        Each item is a dictionary with the keys ``StatementName`` and ``LastModifiedTime``.
+    """
+    athena_client = _utils.client("athena", session=boto3_session)
+    workgroup = workgroup if workgroup else "primary"
+
+    response = athena_client.list_prepared_statements(WorkGroup=workgroup)
+    statements = response["PreparedStatements"]
+
+    while "NextToken" in response:
+        response = athena_client.list_prepared_statements(WorkGroup=workgroup, NextToken=response["NextToken"])
+        statements += response["PreparedStatements"]
+
+    return cast(List[Dict[str, Any]], statements)
+
+
+@apply_configs
+def deallocate_prepared_statement(
```

Comment:
Was this initially done via `DEALLOCATE` SQL query? It's a nitpick but if we're using `DeletePreparedStatement` boto3 API call, I would prefer to method be named after that

Reply:
Yes, that's exactly why it's named this way :)

Good catch! I'll change it.